Looks to be a doctest error

```
=================================== FAILURES ===================================
_____________ [doctest] pandas.core.generic.NDFrame.convert_dtypes _____________
6466         Convert the DataFrame to use best possible dtypes.
6467 
6468         >>> dfn = df.convert_dtypes()
6469         >>> dfn
6470            a  b      c     d     e      f
6471         0  1  x   True     h    10   <NA>
6472         1  2  y  False     i  <NA>  100.5
6473         2  3  z   <NA>  <NA>    20  200.0
6474 
6475         >>> dfn.dtypes
Differences (unified diff with -expected +actual):
    @@ -1,7 +1,7 @@
    -a      Int32
    -b     string
    -c    boolean
    -d     string
    -e      Int64
    -f    Float64
    +a             Int32
    +b    string[python]
    +c           boolean
    +d    string[python]
    +e             Int64
    +f           Float64
     dtype: object
```

--- a/pandas/core/generic.py
+++ b/pandas/core/generic.py
@@ -6473,12 +6473,12 @@ class NDFrame(PandasObject, indexing.IndexingMixin):
         2  3  z   <NA>  <NA>    20  200.0
 
         >>> dfn.dtypes
-        a      Int32
-        b     string
-        c    boolean
-        d     string
-        e      Int64
-        f    Float64
+        a             Int32
+        b    string[python]
+        c           boolean
+        d    string[python]
+        e             Int64
+        f           Float64
         dtype: object
 
         Start with a Series of strings and missing data represented by ``np.nan``.

--- a/pandas/tests/io/formats/test_to_string.py
+++ b/pandas/tests/io/formats/test_to_string.py
@@ -5,7 +5,7 @@ from textwrap import dedent
 import numpy as np
 import pytest
 
-from pandas.compat import pa_version_under6p0
+import pandas.util._test_decorators as td
 
 from pandas import (
     DataFrame,
@@ -342,11 +342,9 @@ def test_to_string_max_rows_zero(data, expected):
     assert result == expected
 
 
+@td.skip_if_no("pyarrow")
 def test_to_string_string_dtype():
     # GH#50099
-    if pa_version_under6p0:
-        pytest.skip()
-
     df = DataFrame({"x": ["foo", "bar", "baz"], "y": ["a", "b", "c"], "z": [1, 2, 3]})
     df = df.astype(
         {"x": "string[pyarrow]", "y": "string[python]", "z": "int64[pyarrow]"}

